basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/referential/component/CityListPanel.java

```diff
@@ -0,0 +1,128 @@
+package fr.openwide.core.basicapp.web.application.referential.component;
+
+import org.apache.wicket.Component;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.form.AjaxFormComponentUpdatingBehavior;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.TextField;
+import org.apache.wicket.markup.html.panel.Fragment;
+import org.apache.wicket.model.ResourceModel;
+import org.odlabs.wiquery.core.events.StateEvent;
+
+import fr.openwide.core.basicapp.core.business.referential.model.City;
+import fr.openwide.core.basicapp.core.business.referential.model.search.CitySort;
+import fr.openwide.core.basicapp.core.util.binding.Bindings;
+import fr.openwide.core.basicapp.web.application.referential.form.AbstractGenericListItemPopup;
+import fr.openwide.core.basicapp.web.application.referential.form.CityPopup;
+import fr.openwide.core.basicapp.web.application.referential.model.CityDataProvider;
+import fr.openwide.core.jpa.more.business.generic.model.EnabledFilter;
+import fr.openwide.core.wicket.more.markup.html.factory.AbstractParameterizedComponentFactory;
+import fr.openwide.core.wicket.more.markup.html.feedback.FeedbackUtils;
+import fr.openwide.core.wicket.more.markup.html.form.EnumDropDownSingleChoice;
+import fr.openwide.core.wicket.more.markup.html.form.LabelPlaceholderBehavior;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.DecoratedCoreDataTablePanel.AddInPlacement;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.DataTableBuilder;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IAddedCoreColumnState;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IDecoratedBuildState;
+
+public class CityListPanel extends AbstractGenericListItemListPanel<City, CitySort, CityDataProvider> {
```

Comment:
Following the changes regarding the popup, the dataProvider and the form panel, I think you should be able to rename this to `AbstractSimpleGenericListItemPanel` with only the `getNewInstance()` method abstract (or even `SimpleGenericListItemPanel` with a `Supplier` as a constructor argument).

Then you won't even need a `CityListPanel` class, as you will just instantiate the `AbstractSimpleGenericListItemPanel` (or `SimpleGenericListItemPanel`) in `ReferentialPage`.